src/Components/DraggableResizer/DraggableResizer.tsx

```diff
@@ -1,156 +1,92 @@
 // Libraries
-import React, {Component, RefObject} from 'react'
+import React, {FunctionComponent, createRef, useState, useEffect} from 'react'
 import classnames from 'classnames'
 
 // Components
 import {DraggableResizerPanel} from './DraggableResizerPanel'
 import {DraggableResizerHandle} from './DraggableResizerHandle'
 
 // Types
-import {StandardClassProps, Orientation, Gradients} from '../../Types'
+import {StandardFunctionProps, Orientation, Gradients} from '../../Types'
 
 // Styles
 import './DraggableResizer.scss'
 
 // Constants
 const NULL_DRAG = -1
 
-interface Props extends StandardClassProps {
+export interface DraggableResizerProps extends StandardFunctionProps {
   /** Orientation the draggable panels stack in */
   handleOrientation: Orientation
   /** Expects and array of numbers between 0 - 1 */
   handlePositions: number[]
   /** Returns the updated array of numbers between 0 - 1, used to manage state */
   onChangePositions: (positions: number[]) => void
   /** Gradient theme for handle */
-  handleGradient: Gradients
+  handleGradient?: Gradients
 }
 
-interface State {
-  initialized: boolean
-  dragIndex: number
-}
-
-export class DraggableResizer extends Component<Props, State> {
-  public static readonly displayName = 'DraggableResizer'
-
-  public static Panel = DraggableResizerPanel
-
-  public static defaultProps = {
-    handleGradient: Gradients.PastelGothic,
-    testID: 'draggable-resizer',
-  }
-
-  private containerRef: RefObject<HTMLDivElement>
-
-  constructor(props: Props) {
-    super(props)
-
-    this.containerRef = React.createRef()
-
-    this.state = {
-      initialized: false,
-      dragIndex: NULL_DRAG,
+const containerRef = createRef<HTMLDivElement>()
```

Comment:
I think you might run into some trouble on a page with multiple DraggableResizers because of this global scope